packages/stark-ui/src/modules/date-picker/components/date-picker.component.ts

```diff
@@ -21,14 +49,34 @@ const componentName: string = "stark-date-picker";
 	// We need to use host instead of @HostBinding: https://github.com/NationalBankBelgium/stark/issues/664
 	host: {
 		class: componentName
-	}
+	},
+	providers: [
+		{
+			provide: NG_VALUE_ACCESSOR,
+			useExisting: StarkDatePickerComponent,
+			multi: true
+		},
+		{
+			provide: NG_VALIDATORS,
+			useExisting: StarkDatePickerComponent,
+			multi: true
+		}
+	]
 })
-export class StarkDatePickerComponent extends AbstractStarkUiComponent implements OnInit {
+export class StarkDatePickerComponent extends AbstractStarkUiComponent implements OnInit, ControlValueAccessor, Validator {
 	/**
 	 * Source Date to be bound to the datepicker model
 	 */
 	@Input()
-	public date: Date;
+	public get value(): Date | undefined {
+		return this._value;
+	}
+	public set value(value: Date | undefined) {
```

Comment:
Can we move this logic into the `ngOnChanges` instead of defining getter/setter?  The same for the other inputs otherwise the code is confusing as we have to look at many places to find how the changes are handled.

Reply:
This specific one cannot be moved to `ngOnChanges` because it is used by internal methods which manipulates the value too: 
  * `public writeValue(obj: any): void`
  * `public onDateChange(event: MatDatepickerInputEvent<moment.Moment>): void`
  * `@Input value: Date | null`

For the others (which are not manipulated by StarkDatePicker) : 
  * dateFilter
  * disabled
  * placeholder
  * maskConfig
  * maxDate
  * minDate
  * required

I'm taking a look on them to see which one I can move to `ngOnChanges` 😊